crates/config/src/sections/upstream_oauth2.rs

```diff
@@ -211,6 +245,10 @@ pub struct LocalpartImportPreference {
     /// If not provided, the default template is `{{ user.preferred_username }}`
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub template: Option<String>,
+
+    /// How to handle conflicts on the claim, default value is `Fail``
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub on_conflict: Option<OnConflict>,
```

Comment:
```suggestion
    #[serde(default, skip_serializing_if = "OnConflict::is_default")]
    pub on_conflict: OnConflict,
```

and define a 'is_default' method on OnConflict